src/Jackett.Common/Indexers/Corsarored.cs

```diff
@@ -1,256 +1,247 @@
+			AddCategoryMapping(13, TorznabCatType.Books, "Textbooks");
+			AddCategoryMapping(14, TorznabCatType.TVDocumentary, "Documentaries");
+			AddCategoryMapping(21, TorznabCatType.AudioVideo, "Music Video");
+			AddCategoryMapping(22, TorznabCatType.TVSport, "Sport");
+			AddCategoryMapping(23, TorznabCatType.TV, "Theater");
+			AddCategoryMapping(24, TorznabCatType.TV, "Wrestling");
+			AddCategoryMapping(25, TorznabCatType.OtherMisc, "Other");
+			AddCategoryMapping(26, TorznabCatType.Console, "Xbox Games");
+			AddCategoryMapping(27, TorznabCatType.Other, "Wallpaper");
+			AddCategoryMapping(28, TorznabCatType.ConsoleOther, "Other Games");
+			AddCategoryMapping(29, TorznabCatType.TV, "TV Series");
+			AddCategoryMapping(30, TorznabCatType.BooksComics, "Comics");
+			AddCategoryMapping(31, TorznabCatType.TV, "TV");
+			AddCategoryMapping(32, TorznabCatType.Console, "Nintendo Games");
+			AddCategoryMapping(34, TorznabCatType.AudioAudiobook, "Audiobook");
+			AddCategoryMapping(35, TorznabCatType.Audio, "Podcasts");
+			AddCategoryMapping(36, TorznabCatType.BooksMagazines, "Newspapers");
+			AddCategoryMapping(37, TorznabCatType.PCPhoneOther, "Phone Apps");
+		}
+
+		private string ApiLatest => $"{SiteLink}api/latests";
+		private string ApiSearch => $"{SiteLink}api/search";
+
+		public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+		{
+			base.LoadValuesFromJson(configJson);
+			var releases = await PerformQuery(new TorznabQuery());
+
+			await ConfigureIfOK(string.Empty, releases.Any(),
+				() => throw new Exception("Could not find release from this URL."));
+
+			return IndexerConfigurationStatus.Completed;
+		}
+
+		private dynamic CheckResponse(WebClientStringResult result)
+		{
+			try
+			{
+				var json = JsonConvert.DeserializeObject<dynamic>(result.Content);
+
+				switch (json)
+				{
+					case JObject _ when json["ok"] != null && (bool) json["ok"] == false:
+						throw new Exception("Server error");
+					default:
+						return json;
+				}
+			}
+			catch (Exception e)
+			{
+				logger.Error("checkResponse() Error: ", e.Message);
+				throw new ExceptionWithConfigData(result.Content, configData);
+			}
+		}
+
+		private async Task<dynamic> SendApiRequest(IEnumerable<KeyValuePair<string, string>> data)
+		{
+			//var jsonData = JsonConvert.SerializeObject(data);
+			var result = await PostDataWithCookiesAndRetry(ApiSearch, data, null, SiteLink, _apiHeaders, null, true);
+			return CheckResponse(result);
+		}
+
+		private async Task<dynamic> SendApiRequestLatest()
+		{
+			var result = await RequestStringWithCookiesAndRetry(ApiLatest, null, SiteLink, _apiHeaders);
+			return CheckResponse(result);
+		}
+
+		private string GetApiCategory(TorznabQuery query)
+		{
+			var cats = MapTorznabCapsToTrackers(query);
+			var apiList = new List<string>();
+			foreach (var apiCat in cats.Select(cat =>
+				_apiCategories.FirstOrDefault(kvp => kvp.Value.Contains(cat)).Key))
```

Comment:
I didn't review the category logic. Just make sure you are not breaking anything.

Reply:
Both searches with "all" (no category) and searches with categories now work as expected which was the bug. I probably could have just ignored categories in the search and returned "all" in every search, but I thought this would be better overall, since it should remove unrelated searches.